docs/Halogen-Component.md

```diff
@@ -47,15 +46,23 @@ See the `Halogen.Signal` documentation.
 #### `component'`
 
 ``` purescript
-component' :: forall p m node req res i. SF1 (Either i req) (node p (m (Either i res))) -> Component p m node req res
+component' :: forall p m req res i. SF1 (Either i req) (HTML p (m (Either i res))) -> Component p m req res
 ```
 
 A variant of `component` which creates a component with some internal, hidden input type.
 
+#### `widget`
+
+``` purescript
+widget :: forall eff i s m. (Functor m) => String -> String -> ((i -> Eff eff Unit) -> Eff eff { node :: Node, state :: s }) -> (i -> s -> Node -> Eff eff (Maybe Node)) -> (s -> Node -> Eff eff Unit) -> Component (Widget eff i) m i i
```

Comment:
Could a record help clean this up? Also hoping we can find a way to remove these `Eff` callbacks through Aff-level MVar / Stream at some point.